connectors/src/connectors/snowflake/lib/utils.ts

```diff
@@ -86,8 +97,19 @@ export const getConnectorAndCredentials = async ({
     logger.error({ connectorId }, "Failed to retrieve credentials");
     return new Err(Error("Failed to retrieve credentials"));
   }
+  // Narrow the type of credentials to just the username/password variant
+  const credentials = credentialsRes.value.credential.content;
+  if ("apiKey" in credentials) {
```

Comment:
this is brittle + has nothing to do with snowflake. This should use a proper type guard